src/tribler/core/components/metadata_store/restapi/channels_endpoint.py

```diff
@@ -395,11 +397,11 @@ async def add_torrent_to_channel(self, request):
                 tdef = TorrentDef.load_from_memory(data)
             elif uri.startswith("magnet:"):
                 _, xt, _ = parse_magnetlink(uri)
-                if (
-                        xt
-                        and is_infohash(codecs.encode(xt, 'hex'))
-                        and (self.mds.torrent_exists_in_personal_channel(xt) or channel.copy_torrent_from_infohash(xt))
-                ):
+
+                if not xt or not is_infohash(codecs.encode(xt, 'hex')):
```

Comment:
**nitpick:** I think it is better to move the `is_infohash` check into the `parse_magnetlink` function, so the function returns an xt value only when it is a correct infohash. Then you can just check here: `if not xt: ...`